src/utils/dom.ts

```diff
@@ -159,10 +159,11 @@ export async function focusElement(el: CalciteFocusableElement): Promise<void> {
 interface GetSlottedOptions {
   all?: boolean;
   direct?: boolean;
+  matches?: string;
```

Comment:
@jcfranco what do you think? Should we have a way to do `getSlotted(this.el, { matches: "calcite-link"});`

Reply:
I think `selector` should work here. Let me know if it doesn't _match_ your needs (badumtss🥁) .

Reply:
@jcfranco selector queries the element for children, matches tests the element.

Reply:
the matches is more of a filter.

Reply:
Gotcha. Looks like you found a hole in this util. 🍩  I think queries using `selector` should include the slotted element. WDYT? Fortunately, there are only 2 selector usages so far:

1. https://github.com/Esri/calcite-components/blob/2ec984876f33939d06c7ad0b47c9040f1a7d0a9d/src/components/calcite-tree-item/calcite-tree-item.tsx#L414
2. https://github.com/Esri/calcite-components/blob/5e0f9f259d160f7826dbf1be56b05ad52b0d0add/src/components/calcite-alert/calcite-alert.tsx#L241

I'll create an issue for the above and can tackle this separately.

Reply:
Yeah we can either update how selector works or matches.